src/main.py

```diff
@@ -1,8 +1,64 @@
-from fastapi import FastAPI
+import uuid
+
+from fastapi import FastAPI, HTTPException
+from src.schemas import ProjectDetails, Project
 
 app = FastAPI()
+projects: list[ProjectDetails] = []
+
+
+def get_project(project_id: uuid.UUID) -> ProjectDetails:
+    project = next((p for p in projects if p.project_id == project_id), None)
+    if project is None:
+        raise HTTPException(status_code=404, detail=f"ProjectDetails {project_id} not found")
+    return project
+
+
+def create_project_(project: Project) -> ProjectDetails:
+    new_project = ProjectDetails(**project.model_dump())
+    projects.append(new_project)
+    return new_project
+
+
+def get_projects_() -> list[ProjectDetails]:
+    return projects
+
+
+def update_project_details_(project_id: uuid.UUID, name: str, description: str | None) -> ProjectDetails:
+    project = get_project(project_id)
+    project.description = description
+    project.name = name
+    return project
+
+
+def delete_project_(project_id: uuid.UUID) -> ProjectDetails:
+    for project in projects:
+        if project.project_id == project_id:
+            projects.remove(project)
+            return project
```

Comment:
1. function received an object (instance) of a project to delete
2. function removes this project

You already implemented get_project function, please use that in your router, pass it to the delete function, simple as that